File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-3.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int counter = 1;
        if (nums.length == 0) {
            return 0;
        }

        for(int i = 0; i < nums.length-1; i++){
            if(nums[i] == nums[i+1]){
                nums[counter] = nums[i];
            }else{
                nums[counter] = nums[i+1];
                counter++;
            }
        }
        return counter;
    }
}

File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-3.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        left = 0
        right = len(nums)-1

        while(left<right):
            mid = left + (right-left)//2
            if(nums[mid]<nums[right]):
                right = mid
            else:
                left = mid + 1

        pivot = left

        if(nums[pivot] <= target <= nums[-1]):
            return self.binarySearch(nums, target, pivot, len(nums)-1)
        else:
            return self.binarySearch(nums, target, 0, pivot-1)

    def binarySearch(self, nums: List[int], target: int, left: int, right: int) -> int:
        while(left<=right):
            mid = left + (right - left)//2
            if(nums[mid]<target):
                left = mid + 1
            elif(nums[mid]>target):
                right = mid - 1
            elif(nums[mid]==target):
                return mid
        return -1
